app/domain/people/household_list.rb

```diff
@@ -37,6 +38,20 @@ def grouped_households
       .select("people.*")
       .from(grouped_households_people_sql, :people)
       .limit(@people_scope.limit_value.presence)
+      .order(order_statement)
+  end
+
+  def order_statement
+    case @order
+    when :retain
+      quoted_ids = @people_scope.pluck(:id).map { |id| Arel::Nodes.build_quoted(id).to_sql }
+      array_literal = Arel.sql("ARRAY[#{quoted_ids.join(",")}]")
+      Arel::Nodes::NamedFunction.new("array_position", [array_literal, Person.arel_table[:id]])
+    when :default
+      '"member_count" DESC, id ASC'
+    else
+      @order
+    end
   end
```

Comment:
Meine Idee war hier, die bestehende Reihenfolge einfach beizubehlaten, indem mit einem Array-Literal sortiert wird. Performance-mässig sollte das trotzdem gut gehen, aber Arel unterstützt die ARRAY[]-Literals von Postgres leider nicht. Was denkst du @kronn

Reply:
Der Ansatz, erst die Sortier-Reihenfolge zu generieren und dann anzuwenden, ist IMHO okay. `ORDER BY array_position` nehme ich mal in meine persönliche Toolbox. Da die Exporte meist im Background-Job laufen, ist Performance nicht die erste Priorität. Ich gehe aber davon aus, dass es nicht extrem anders ist als den `member_count` zu nutzen.

Der fehlende Support für Array-Typen von PostgreSQL in Rails... ja... normal? Ich finde es aber gut, dass `Arel::Nodes` verwendet werden, die ermöglichen, die Query möglichst spät und möglichst nur einmal zu evaluieren.